newsroom/commands/remove_expired_agenda.py

```diff
@@ -0,0 +1,179 @@
+
+        if len(items_to_remove):
+            logger.info(f"Deleting items: {items_to_remove}")
+            num_items_removed += len(items_to_remove)
+            agenda_service.delete_action(lookup={config.ID_FIELD: {"$in": list(items_to_remove)}})
+
+    logger.info("Finished removing expired items from agenda collection")
+    return num_items_removed
+
+
+def _get_expired_items(expiry_datetime: datetime) -> Generator[List[Dict[str, Any]], datetime, None]:
+    """Get the expired items, based on ``expiry_datetime``"""
+
+    agenda_service = get_resource_service("agenda")
+    max_loops = app.config.get("MAX_EXPIRY_LOOPS", 50)
+    ids_processed: Set[str] = set()
+    for i in range(max_loops):  # avoid blocking forever just in case
+        req = ParsedRequest()
+        expiry_datetime_str = date_to_str(expiry_datetime)
+
+        # Filters out Planning items with coverages that have not yet expired
+        coverage_scheduled_query = {
+            "nested": {
+                "path": "coverages",
+                "query": {"range": {"coverages.scheduled": {"gt": expiry_datetime_str}}},
+            },
+        }
+        req.args = {
+            "source": json.dumps({
+                "query": {
+                    "bool": {
+                        "must": [{"range": {"dates.end": {"lte": expiry_datetime_str}}}],
+                        "should": [
+                            # Match Events directly (stored from v2.3+)
+                            # No more filters required, as we'll query & check planning items separately
+                            {"term": {"item_type": "event"}},
+
+                            # Match Planning directly with no associated Event (stored from v2.3+)
+                            {"bool": {
+                                "must": [{"term": {"item_type": "planning"}}],
+                                "must_not": [
+                                    {"exists": {"field": "event_id"}},
+                                    coverage_scheduled_query,
+                                ],
+                            }},
+
+                            # Match Event and/or Planning items (stored before v2.3 changes to storage)
+                            {"bool": {
+                                "must_not": [
+                                    {"exists": {"field": "item_type"}},
+                                    coverage_scheduled_query,
+                                ],
+                            }},
+                        ],
+                        "minimum_should_match": 1,
+                    },
+                },
+                "sort": [{"dates.start": "asc"}],
+                "size": app.config.get("MAX_EXPIRY_QUERY_LIMIT", 100),
+            }),
+        }
+
+        items = list(agenda_service.internal_get(req=req, lookup=None))
+
+        if not len(items):
+            break
+
+        for item in items:
+            ids_processed.add(item[config.ID_FIELD])
```

Comment:
not sure what's the use for this

Reply:
Ah, forgot that one. I re-wrote that es query multiple times.
I should use that and add it into the query so the next request does not include the items already processed. Issue here though is the query may become too big if I keep adding in these IDs.
Will fix it tomorrow.

Reply:
@petrjasek I removed this variable entirely. Considering we're talking about Events & their associated Coverages, I don't foresee a case where the Coverage would be 60+ days after the Event's scheduled end date (based on example `AGENDA_EXPIRY_DAYS` config).